eng/common/pipelines/templates/steps/set-test-pipeline-version.yml

```diff
@@ -4,7 +4,7 @@ parameters:
   TestPipeline: false
 
 steps:
-- ${{if eq(parameters.TestPipeline, 'true')}}:
+- ${{if and(eq(parameters.TestPipeline, 'true'), ne(variables['Skip.SetTestPipelineVersion'], 'true')) }}:
```

Comment:
Are you sure this works? I cannot remember for sure but I don't know if queue time variables can be used in template conditions. I believe that is why we had to make them all runtime conditions in our release stages.